services/galley/test/integration/API/Teams.hs

```diff
@@ -294,28 +311,27 @@ testDeleteTeam g b c = do
 
     void $ WS.bracketR3 c owner extern (member^.userId) $ \(wsOwner, wsExtern, wsMember) -> do
         delete (g . paths ["teams", toByteString' tid] . zUser owner . zConn "conn") !!!
-            const 200 === statusCode
-
-        get (g . paths ["teams", toByteString' tid] . zUser owner) !!!
-            const 404 === statusCode
-
-        get (g . paths ["teams", toByteString' tid, "members"] . zUser owner) !!!
-            const 403 === statusCode
-
-        get (g . paths ["teams", toByteString' tid, "conversations"] . zUser owner) !!!
-            const 403 === statusCode
-
-        for_ [owner, extern, member^.userId] $ \u ->
-            for_ [cid1, cid2] $ \x -> do
-                Util.getConv g u x !!! const 404 === statusCode
-                Util.getSelfMember g u x !!! do
-                    const 200         === statusCode
-                    const (Just Null) === Util.decodeBody
-
+            const 202 === statusCode
         checkTeamDeleteEvent tid wsOwner
         checkTeamDeleteEvent tid wsMember
         checkConvDeletevent  cid1 wsExtern
         WS.assertNoEvent timeout [wsOwner, wsExtern, wsMember]
+
+    get (g . paths ["teams", toByteString' tid] . zUser owner) !!!
+        const 404 === statusCode
```

Comment:
Given the async nature of team deletion (and that events are sent out before removal from the DB), this could eventually still return a `200` in some cases, right? And if so, would it make sense to retry a few times then?